core/src/main/kotlin/com/malinskiy/marathon/io/FileManager.kt

```diff
@@ -65,8 +68,15 @@ class FileManager(private val output: File) {
     private fun createFilename(fileType: FileType, testBatchId: String): String =
         "$testBatchId.${fileType.suffix}"
 
-    private fun createFilename(test: Test, fileType: FileType, testBatchId: String? = null): String =
-        "${test.toTestName()}${testBatchId?.let { "-$it" } ?: ""}.${fileType.suffix}"
+    private fun createFilename(test: Test, fileType: FileType, testBatchId: String? = null): String {
+        val testSuffix = "${testBatchId?.let { "-$it" } ?: ""}.${fileType.suffix}"
+        val testName = test.toTestName().take(256 - testSuffix.length)
```

Comment:
What about conflicts? For example: We have two tests with very long names with difference only in last character, after we take `256 - testSuffix.length` they will have equal names. I think in this case it makes more sense to use hash or uuid instead of shortening names.

Reply:
We can either make this human-readable or unique

Reply:
I would say correct or human-readable :)

Reply:
I can see your preference here :) I don't think we can remove the human-readable part though - people sometimes browse the files and they would be completely useless if they're just UUIDs

Reply:
I think having correct and human-readable filenames is possible, but maybe a bit complicated. Calculating diff between two test names and using it for smart shortening algorithm may work, but sounds too complicated. I think we can go with the existing solution, but also add logs(with error level) in case we have a conflict.